plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/guru/VMwareGuru.java

```diff
@@ -1104,13 +1124,22 @@ public List<Command> finalizeMigrate(VirtualMachine vm, Map<Volume, StoragePool>
             if (pool.getClusterId() != null) {
                 poolClusterId = pool.getClusterId();
             }
+            if (volume.getVolumeType().equals(Volume.Type.ROOT) && pool.isLocal()) {
+                isLocalStorageMigration = true;
+            }
             volumeToFilerTo.add(new Pair<VolumeTO, StorageFilerTO>(volumeTo, filerTo));
         }
         final Long destClusterId = poolClusterId;
         final Long srcClusterId = vmManager.findClusterAndHostIdForVm(vm.getId()).first();
         final boolean isInterClusterMigration = isInterClusterMigration(destClusterId, srcClusterId);
+        String targetHostGuid = null;
+        if (isLocalStorageMigration) {
```

Comment:
```suggestion
        if (!isLocalStorageMigration || targetHostGuid == null) {
            targetHostGuid = getHostGuidInTargetCluster(isInterClusterMigration, destClusterId);
        }
```

@harikrishna-patnala See if `targetHostGuid == null` cond valid there or not (as getHostGuidForLocalStorage() can return null in some cases).

Reply:
Ideally, targetHostGuid in case of local storage should not return null. Also I have not changed the code in the before review comment, so I would like to keep it as is. Please suggest.